src/components/implementation/tests/micro_booter/mb_tests.c

```diff
@@ -499,267 +453,299 @@ test_budgets(void)
+	if (EXPECT_LL_NEQ(0, strcmp(chk, p), "Memory Test")) {
+		return;
+	}
 
-	while (1) {
-		seq_order_check(e);
-		if (wakeup_budget_test[cos_cpuid()]) {
-			rdtscll(e->cyc);
-			while (1)
-				;
-		} else {
-			cos_switch(BOOT_CAPTBL_SELF_INITTHD_CPU_BASE, BOOT_CAPTBL_SELF_INITTCAP_CPU_BASE, 0, 0, 0, 0);
+	s = cos_page_bump_alloc(&booter_info);
+	assert(s);
+	prev = s;
+	for (i = 0; i < TEST_NPAGES; i++) {
+		t = cos_page_bump_alloc(&booter_info);
+		if (t == NULL){
+			EXPECT_LL_EQ(0, 1, "Memory Test");
+			return;
 		}
+		if (t != prev + PAGE_SIZE) {
+			fail_contiguous = 1;
+		}
+		prev = t;
+	}
+	if (!fail_contiguous) {
+		memset(s, 0, TEST_NPAGES * PAGE_SIZE);
+	} else if (EXPECT_LL_EQ(i, TEST_NPAGES,"Memory Test\tallocate contiguous")) {
+		return;
+	}
+
+	t = cos_page_bump_allocn(&booter_info, TEST_NPAGES * PAGE_SIZE);
+	if (t == NULL) {
+		EXPECT_LL_NEQ(0, 1, "Memory Test");
+		return;
 	}
+	memset(t, 0, TEST_NPAGES * PAGE_SIZE);
 }
 
-/* intr thread - interrupt thread */
+volatile arcvcap_t rcc_global[NUM_CPU], rcp_global[NUM_CPU];
+volatile asndcap_t scp_global[NUM_CPU];
+int                async_test_flag_[NUM_CPU] = { 0 };
+
 static void
-intr_thd(void *d)
+async_thd_fn_perf(void *thdcap)
 {
-	struct exec_cluster *e = &(((struct activation_test_data *)d)->i);
-	struct exec_cluster *w = &(((struct activation_test_data *)d)->w);
+	thdcap_t  tc = (thdcap_t)thdcap;
+	arcvcap_t rc = rcc_global[cos_cpuid()];
+	int       i;
 
-	while (1) {
-		cos_rcv(e->rc, 0, NULL);
-		seq_order_check(e);
-		cos_thd_wakeup(w->tc, w->tcc, w->prio, wakeup_budget_test[cos_cpuid()] ? TEST_WAKEUP_BUDGET : 0);
+	cos_rcv(rc, 0, NULL);
+
+	for (i = 0; i < ITER + 1; i++) {
+		cos_rcv(rc, 0, NULL);
 	}
+
+	cos_thd_switch(tc);
 }
 
-/* scheduler of the intr_thd */
 static void
-intr_sched_thd(void *d)
+async_thd_parent_perf(void *thdcap)
 {
-	struct exec_cluster *e = &(((struct activation_test_data *)d)->s);
-	cycles_t             cycs;
-	int                  blocked;
-	thdid_t              tid;
-	tcap_time_t          thd_timeout;
+	thdcap_t  tc                = (thdcap_t)thdcap;
+	asndcap_t sc                = scp_global[cos_cpuid()];
+	long long total_asnd_cycles = 0;
+	long long start_asnd_cycles = 0, end_arcv_cycles = 0;
+	int       i;
 
-	while (1) {
-		cos_sched_rcv(e->rc, 0, 0, NULL, &tid, &blocked, &cycs, &thd_timeout);
-		seq_order_check(e);
-		if (wakeup_budget_test[cos_cpuid()]) {
-			struct exec_cluster *w = &(((struct activation_test_data *)d)->w);
-			rdtscll(e->cyc);
-			printc(" | preempted worker @ %llu, budget: %lu=%lu |", e->cyc,
-			       (unsigned long)TEST_WAKEUP_BUDGET, (unsigned long)(e->cyc - w->cyc));
-		}
+	cos_asnd(sc, 1);
+
+	rdtscll(start_asnd_cycles);
+	for (i = 0; i < ITER; i++) {
+		cos_asnd(sc, 1);
 	}
+	rdtscll(end_arcv_cycles);
+	total_asnd_cycles = (end_arcv_cycles - start_asnd_cycles) / 2;
+
+	PRINTC("Average ASND/ARCV (Total: %lld / Iterations: %lld ): %lld\n", total_asnd_cycles, (long long)(ITER),
```

Comment:
Perhaps move Performance tests out of this file to something like `mb_perf_tests.c`. Have two files in the same folder or have 2 separate folders.